### PR TITLE
improve UX regarding target time

### PR DESCRIPTION
**Pull request by szaimen**
This is best reviewed like this https://github.com/nextcloud/calendar/pull/3825/files?diff=unified&w=1

Close https://github.com/nextcloud/calendar/issues/3824

## Before
https://user-images.githubusercontent.com/42591237/146625622-af0640fd-7da5-4d03-acc0-cde6fbbd3d74.mp4

## After
https://user-images.githubusercontent.com/42591237/146625491-3b64b552-e0f5-4ee8-8c19-beb51d4785c7.mp4

Signed-off-by: szaimen <szaimen@e.mail.de>

<details>
<summary>For my own testing</summary>

```
docker run -it \
-e CALENDAR_BRANCH=enh/3824/target-time \
-p 8443:443 \
-e TRUSTED_DOMAIN=192.168.146.128 \
--name nextcloud-easy-test \
ghcr.io/szaimen/nextcloud-easy-test:latest
```

</details>

### CODE REVIEW

**Comment by jancborchardt** [comment]
In the "after" video, the calendar grid jumps after clicking though?

**Comment by szaimen** [comment]
> In the "after" video, the calendar grid jumps after clicking though?

Yes, the jumping is another issue and not related to this change :)

**Review by jancborchardt — APPROVED** [review]
Change looks good, nice improvement :)

File: src/store/calendarObjectInstance.js [hunk 1]
```diff
@@ -215,10 +215,17 @@ const mutations = {
 			calendarObjectInstance.eventComponent.endDate.addDuration(DurationValue.fromSeconds(60 * 60 * 24))
 		} else {
 			if (endDateObj.compare(startDateObj) === -1) {
-				const timezone = getTimezoneManager().getTimezoneForId(startDateObj.timezoneId)
-				calendarObjectInstance.eventComponent.startDate
-					= calendarObjectInstance.eventComponent.endDate.getInTimezone(timezone)
-				calendarObjectInstance.startDate = getDateFromDateTimeValue(calendarObjectInstance.eventComponent.startDate)
+				endDateObj.addDuration(DurationValue.fromSeconds(60 * 60 * 24))
```
Comment (st3iny) [inline src/store/calendarObjectInstance.js:218]:
```suggestion
				// Is end is more than one day before start?
				endDateObj.addDuration(DurationValue.fromSeconds(60 * 60 * 24))
```

File: src/store/calendarObjectInstance.js [hunk 2]
```diff
@@ -215,10 +215,17 @@ const mutations = {
 			calendarObjectInstance.eventComponent.endDate.addDuration(DurationValue.fromSeconds(60 * 60 * 24))
 		} else {
 			if (endDateObj.compare(startDateObj) === -1) {
```
Comment (st3iny) [inline src/store/calendarObjectInstance.js:217]:
```suggestion
			// Is end before start?
			if (endDateObj.compare(startDateObj) === -1) {
```

**Review by st3iny — APPROVED** [review]
Very nice usability improvement. 

I left some annotations to add some comments in the code. The duplicated if statement might be hard to understand later on.

**Comment by szaimen** [comment]
🎉

**Pull request merged**